backend/src/services/certificate-authority/internal/internal-certificate-authority-fns.ts

```diff
@@ -257,7 +260,274 @@ export const InternalCertificateAuthorityFns = ({
     };
   };
 
+  const issueCertificateWithTemplate = async (
```

Comment:
Nit: Significant overlap with `issueCertificate` function; would be great to somehow reuse that function / feed that function into this one / apply some way to streamline similar logic so in the future we don't have situation where we update in one place and forget to update in another place.

Ideally updating in one place would also apply elsewhere as well with related functionality that is dependent or similar to it.

Can do it in this or separate PR.